### ISSUE TITLE
Allow users to mark and/or sort their downloads

### ISSUE THREAD

**Issue by gbif-portal** [gbif/portal-feedback#5214]
**Permanent removal of downloads**

Hi! I have requested a number of downloads over the past 3-4 years, but I only end up using a few of them for research. Therefore, the "Download" tab of my user is cluttered by irrelevant downloads. For instace, some of them were mere tests of how rgbif works. I would like to permanently remove such downloads, so I can focus on the downloads that actually mattered for research. Is there a way to remove the downloads from view?
Thanks!

-----

Github user: @MarcRieraDominguez
User: [See in registry](https://www.gbif.org/api/feedback/user/0eddebe9ce54d2c4eae8e1f58d4b646a:5d22293d18e8312764accef78988268aee747355bba774e85c5d6268baee851bd1d3c00999acae27f16c43adba04001043cb4c91a844c9eb04d2afb58d684e3f9ad245abdce64dcb1dd4347f048e645b0b73f43eabea0cf39899e04cde435d6352cfa47ab7d2e17b7c4ac3b9900a491aee0aec012d1ba261af6d454002a2cb8a) - [Send email](https://www.gbif.org/api/feedback/user/mailto/0eddebe9ce54d2c4eae8e1f58d4b646a:5d22293d18e8312764accef78988268aee747355bba774e85c5d6268baee851bd1d3c00999acae27f16c43adba04001043cb4c91a844c9eb04d2afb58d684e3f9ad245abdce64dcb1dd4347f048e645b0b73f43eabea0cf39899e04cde435d6352cfa47ab7d2e17b7c4ac3b9900a491aee0aec012d1ba261af6d454002a2cb8a)
System: Chrome 122.0.0 / Windows 10.0.0
Referer: https://www.gbif.org/faq?question=for-how-long-will-does-gbif-store-downloads
Window size: width 1920 - height 911
[API log](https://private-logs.gbif.org/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2024-02-29T09:58:13.310Z',to:'2024-02-29T10:04:13.310Z'))&_a=(columns:!(_source),filters:!(),index:'3390a910-fcda-11ea-a9ab-4375f2a9d11c',interval:auto,query:(language:kuery,query:''),sort:!()))
[Site log](https://private-logs.gbif.org/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2024-02-29T09:58:13.310Z',to:'2024-02-29T10:04:13.310Z'))&_a=(columns:!(_source),filters:!(),index:'5c73f360-fce3-11ea-a9ab-4375f2a9d11c',interval:auto,query:(language:kuery,query:''),sort:!()))
System health at time of feedback: OPERATIONAL

**Comment by ManonGros** [comment]
Hi @MarcRieraDominguez

We don’t delete pages that are associated with DOIs. DOIs must have a landing page, they must remain persistent.
The button to delete downloads only delete the occurrence file associated with the download. The page of the download with the data about the download (query, datasets involved,…) remains accessible via the DOI.

I hope this makes sense.
I will close the issue but let us know if it should be re-opened.

**Comment by MarcRieraDominguez** [comment]
Thank you for your reply! I understand about the "permanence" of DOIs. However, could they be sorted by relevance somehow? My current user profile is cluttred by test downloads. Could the DOIs be sorted in two tabs, according to whether the user marks them as relevant or irrelevant? I am thinking in terms of enhancing the user's control over what is being displayed. Cheers! :)

**Comment by ManonGros** [comment]
Thanks Marc, I reopened the issue so we keep the idea.

**Comment by MortenHofft** [comment]
I can see two fairly easy additions:

1: label the ones eligible for deletion
2: allow filtering by status

**status filter**
We could add a status filter. Such that the user could choose only to see `cancelled`, `succeeded` or `file_erased` downloads? `file_erased` erased is the status assigned once the file is deleted and that happens typically 6 months after it has been marked by the user as eligible for deletion. 

To make it more useful we could add an additional status: `eligible_for_deletion` or such, to indicate that it succeeded, but is flagged to have the file deleted in the future. 

Would that solve it @MarcRieraDominguez ?

**Comment by MortenHofft** [comment]
The UI now shows downloads that has been marked as eligible for deletion. So the red border should make it easy to skip those in a visual scanning. As far as I can see you have 28 downloads.

example
<img width="980" alt="Screenshot 2024-03-18 at 13 02 04" src="https://github.com/gbif/portal-feedback/assets/6759267/bf50969e-cc95-4c3b-8282-2cbce6b82425">

**For testing it is btw better to use our test environment UAT. That will also keep your production downloads less cluttered.**

**Comment by MarcRieraDominguez** [comment]
Thank you @MortenHofft! I see the red border now, and it makes things easier.
Is the idea of filtering by status still on the table? I think it would enhance user experience even further.
I didn't know about this UAT, interesting. Is there a way to make it work from rgbif?

**Comment by MortenHofft** [comment]
@jhnwllr can answer rgbif questions

Yes filtering is still an option, but there is currently no backend/api to filter by downloads labeled as "can be deleted, but not yet deleted". So I will let the issue remain open

**Comment by jhnwllr** [comment]
Yes there is no good way to do this right now. I am working on an eraseAfter feature for rgbif which might help, but I am not sure if the download will get automatically deleted even if this field is set. I think deleting downloads is still a manual process. @MattBlissett

**Comment by jhnwllr** [comment]
@MarcRieraDominguez it's not possible to download from rgbif from UAT, without modifying code. 
You would have to clone a local copy and change this line. 

https://github.com/ropensci/rgbif/blob/master/R/zzz.r#L382

**Comment by MortenHofft** [comment]
Argh, I've misunderstood the API. Apparently all downloads are deleted as default. And also will be if the user clicks postpone. So now everything is marked as eligible for deletion unless a citation is spotted. That makes the UI far more confusing in most cases.

I believe it should be removed again, and possibly be replaced by a reverse tag [has citations and will not be deleted]
The filter to remove deleted files is then the only option we have.
